prow/resultstore/client.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resultstore
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+
+	"google.golang.org/genproto/googleapis/devtools/resultstore/v2"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+)
+
+// TODO: have client connect itself? Move to flagutils?
+const ResultStoreAddress = "resultstore.googleapis.com:443"
```

Comment:
We'd probably make this a flag for integration testing (with this as the default value). My comment was more specifically about simplifying the process of creating a new client so that callers only need `NewClient` and not `Connect`.